src/Controls/tests/BindingSourceGen.UnitTests/ObservablePropertyTests.cs

```diff
@@ -0,0 +1,292 @@
+		var allGeneratedCode = string.Join("\n\n", result.GeneratedFiles.Values);
+		Assert.Contains("Name", allGeneratedCode, System.StringComparison.Ordinal);
+
+		// Note: There will be compilation errors because Name doesn't actually exist,
+		// but the interceptor code itself is generated correctly. In real usage with
+		// CommunityToolkit.Mvvm, the property would exist and compile successfully.
+	}
+
+	[Fact]
+	public void GenerateBindingToObservablePropertyFromUnderscorePrefixedField()
+	{
+		var source = """
+			using Microsoft.Maui.Controls;
+			using System.Collections.ObjectModel;
+
+			namespace CommunityToolkit.Mvvm.ComponentModel
+			{
+				[System.AttributeUsage(System.AttributeTargets.Field)]
+				public class ObservablePropertyAttribute : System.Attribute { }
+			}
+
+			namespace TestApp
+			{
+				public class MyViewModel
+				{
+					[CommunityToolkit.Mvvm.ComponentModel.ObservableProperty]
+					private string? _title;
+					// Title property will be generated by CommunityToolkit.Mvvm
+				}
+
+				public class TestCode
+				{
+					public void Test()
+					{
+						var label = new Label();
+						label.SetBinding(Label.TextProperty, static (MyViewModel vm) => vm.Title);
+					}
+				}
+			}
+			""";
+
+		var result = SourceGenHelpers.Run(source);
+
+		// The binding should be generated successfully with ObservableProperty inference
+		Assert.NotNull(result.Binding);
+
+		// Verify the generated interceptor code contains the Title property access
+		var allGeneratedCode = string.Join("\n\n", result.GeneratedFiles.Values);
+		Assert.Contains("Title", allGeneratedCode, System.StringComparison.Ordinal);
+	}
+
+	[Fact]
+	public void GenerateBindingToObservablePropertyCollection()
+	{
+		var source = """
+			using Microsoft.Maui.Controls;
+			using System.Collections.ObjectModel;
+
+			namespace CommunityToolkit.Mvvm.ComponentModel
+			{
+				[System.AttributeUsage(System.AttributeTargets.Field)]
+				public class ObservablePropertyAttribute : System.Attribute { }
+			}
+
+			namespace TestApp
+			{
+				public class Tag { }
+
+				public class MyViewModel
+				{
+					[CommunityToolkit.Mvvm.ComponentModel.ObservableProperty]
+					private ObservableCollection<Tag> _tags = new();
+					// Tags property will be generated by CommunityToolkit.Mvvm
+				}
+
+				public class TestCode
+				{
+					public void Test()
+					{
+						var label = new Label();
+						label.SetBinding(Label.BindingContextProperty, static (MyViewModel vm) => vm.Tags);
+					}
+				}
+			}
+			""";
+
+		var result = SourceGenHelpers.Run(source);
+
+		// The binding should be generated successfully with ObservableProperty inference
+		Assert.NotNull(result.Binding);
+
+		// Verify the generated interceptor code contains the Tags property access
+		var allGeneratedCode = string.Join("\n\n", result.GeneratedFiles.Values);
+		Assert.Contains("Tags", allGeneratedCode, System.StringComparison.Ordinal);
```

Comment:
Let's improve the assert to check for the full expected getter and setter: `"source => (source.Tags, true)"`, `source.Tags = value;`.

I might be wrong about the exact values of the expected generated expressions, so please double check the actual outputs.